### PR TITLE
Refactor relationship

### PR DESCRIPTION
**Pull request by wenhao-zhang**
The PR is for the refactor of relationships. 

Changes:
-Groups and relationships moving to their own models
-Unused user endpoints going to be removed
-Account creation will also create groups and relationships and associate them to the account

### CODE REVIEW

File: src/controllers/relationship.js [hunk 1]
```diff
@@ -46,12 +46,12 @@ module.exports = [
             throw error;
```
Comment (wenweil) [inline src/controllers/relationship.js:46]: req.auth.id.toString() === friendId.toString()

Isnt this a different error from USER_DOES_NOT_EXIST?

**Comment by wenweil** [comment]
isnt there supposed to be routes for these end points?

File: src/controllers/relationship.js (hunk 1 above)
Comment (wenhao-zhang) [reply src/controllers/relationship.js:46]: It kind of is, but I tried on discord and they seem to pretty much treat it as a not found. That is why I did it here

**Review by wenweil — APPROVED** [review]
Looks good to me

**Pull request merged**